.github/workflows/plugin_update.yml

```diff
@@ -0,0 +1,54 @@
+name: Plugins Update file test
+
+on:
+  schedule:
+    - cron: '0 0 * * *' # run daily at midnight
+
+jobs:
+  check-plugins:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v2
+      
+      - name: Configure git & gh
+        run: |
+          
+          echo ${{ secrets.GH_TOKEN }} | gh auth login --with-token
+          gh auth status
+          git config --global user.email "ash.191245141@gmail.com"
+          git config --global user.name "ash-sxn"
+
+      - name: Check plugins
+        run: |
+          # Set the name of the branch to create the pull request from
+          BRANCH_NAME="update-plugins"
+
+          # Set the name of the branch to create the pull request to
+          BASE_BRANCH="main"
+          
+          # Start the docker containers
+          docker compose up -d 
+          # Uses jenkins-plugin-cli to update the plugin.txt and update it in root
+          docker ps | grep controller | awk '{print $1}' | xargs -I {} docker exec -u root {} jenkins-plugin-cli --plugin-file /usr/share/jenkins/ref/plugins.txt --no-download --available-updates --output txt > dockerfiles/plugins.txt
```

Comment:
Why do you need to use `root` for this?
How do you know your Jenkins instance has finished initializing?
I would have expected the same kind of controls you did in the test GitHub Action. 🤔